src/screens/lists/Lists.tsx

```diff
@@ -0,0 +1,211 @@
+  }, [getAllData]);
+
+  const cardsArr = useMemo(() => {
+    return (listId: string) => cards.filter(({idList}) => idList === listId);
+  }, [cards]);
+
+  const onPressBack = () => {
+    boardNavigation.goBack();
+  };
+
+  const onPressSetting = () => {
+    boardNavigation.navigate(ROUTES.BOARDS_STACK_SCREEN.SETTINGS_SCREEN);
+  };
+
+  const onPressClear = () => {
+    isAddList ? setIsAddList(false) : setAddingCardToListId('');
+  };
+
+  const onPressCheck = async () => {
+    if (isAddList) {
+      await addData(newListUrl(token, newList, boardId));
+      getAllData();
+      setNewList('');
+      setIsAddList(false);
+    } else {
+      await addData(newCardUrl(token, newCard, addingCardToListId));
+      getAllData();
+      setNewCard('');
+      setAddingCardToListId('');
+    }
+  };
+
+  const onInputChange = (value: string, type: 'list' | 'card') => {
+    type === 'list' ? setNewList(value) : setNewCard(value);
+  };
+
+  const onPressCard = (id: string) => {
+    boardNavigation.navigate(ROUTES.BOARDS_STACK_SCREEN.CARD_DETAILS_SCREEN, {
+      cardId: id,
+    });
+  };
+
+  const renderCard = ({item}: {item: CardInfo}) => (
+    <TouchableOpacity onPress={() => onPressCard(item.id)}>
+      <View style={styles.cardContainer}>
+        <Text style={styles.cardTitle}>{item.name}</Text>
+      </View>
+    </TouchableOpacity>
+  );
+
+  const renderList = (list: ListInfo) => (
+    <View key={list.id} style={styles.listContainer}>
+      <View style={styles.listChildContainer}>
+        <Text style={styles.listTitle}>{list.name}</Text>
+        <FlatList data={cardsArr(list.id)} renderItem={renderCard} />
+        {addingCardToListId === list.id ? (
+          <TextInput
+            placeholder="Card name"
+            placeholderTextColor={colors.gray400}
+            autoFocus
+            style={styles.input}
+            onChangeText={value => onInputChange(value, 'card')}
+          />
+        ) : (
+          <Pressable onPress={() => setAddingCardToListId(list.id)}>
+            <Text style={styles.addCard}>+ Add Card</Text>
+          </Pressable>
+        )}
+      </View>
+    </View>
+  );
+
+  const renderAddList = () => (
+    <View style={styles.listContainer}>
+      <View style={styles.listChildContainer}>
+        {isAddList ? (
+          <TextInput
+            placeholder="List name"
+            placeholderTextColor={colors.gray400}
+            autoFocus
+            style={styles.input}
+            onChangeText={value => onInputChange(value, 'list')}
+          />
+        ) : (
+          <Pressable onPress={() => setIsAddList(true)}>
+            <Text style={styles.addList}>Add List</Text>
+          </Pressable>
+        )}
+      </View>
+    </View>
+  );
+
+  const renderHeaderLeftIcon = (isLeft: boolean) => {
+    const iconName =
+      isAddList || addingCardToListId
+        ? 'clear'
+        : Platform.OS === 'android'
+        ? 'arrow-back'
+        : 'arrow-back-ios-new';
```

Comment:
Avoid nested ternary operations, create simple conditions so that if we ever want to debug this condition, we can easily to that. 
Nested ternary operations are hard to understand when it contains more than 2 operations